airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source_adapter.py

```diff
@@ -58,3 +61,45 @@ def _select_abstract_streams(self, config: Mapping[str, Any], configured_catalog
             if isinstance(stream_instance, AbstractStreamFacade):
                 abstract_streams.append(stream_instance.get_underlying_stream())
         return abstract_streams
+
+    def _convert_to_concurrent_stream(
+        self, config: Mapping[str, Any], stream: Stream, state_manager: ConnectorStateManager, cursor: Optional[Cursor] = None
+    ) -> Stream:
+        """
+        Prepares a stream for concurrent processing by initializing or assigning a cursor,
+        managing the stream's state, and returning an updated Stream instance.
+
+        :param config: Configuration parameters necessary for setting up the stream for concurrent processing.
+        :type config: Mapping[str, Any]
+
+        :param stream: The stream object that needs to be prepared for concurrent processing.
+        :type stream: Stream
+
+        :param state_manager: Responsible for managing the storage and retrieval of the stream's state.
+        :type state_manager: ConnectorStateManager
+
+        :param cursor: An optional cursor to manage stream state and processing boundaries.
+        :type cursor: Optional[ConcurrentCursor]
+
+        :return: A Stream instance configured for concurrent processing.
+        :rtype: Stream
+        """
+        logger = logging.getLogger("airbyte")
+
+        if cursor:
+            state = cursor.state
+
+            if hasattr(stream, "set_cursor") and stream.cursor_field:
+                stream.set_cursor(cursor)
+
+            if cursor and hasattr(stream, "parent") and hasattr(stream.parent, "set_cursor"):
+                stream.parent.set_cursor(cursor)
```

Comment:
I feel like this should be done as part of the stream.set_cursor method. I assume that's not possible today because there's no standard way to define parent streams...

Reply:
you are right 
I think we can leave only hasattr(stream, "parent") and set cursor to parent with stream interface in that case